src/Symfony/Component/Serializer/Tests/Context/Encoder/CsvEncoderContextBuilderTest.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Tests\Context\Encoder;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Serializer\Context\Encoder\CsvEncoderContextBuilder;
+use Symfony\Component\Serializer\Encoder\CsvEncoder;
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+
+/**
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ */
+class CsvEncoderContextBuilderTest extends TestCase
+{
+    private CsvEncoderContextBuilder $contextBuilder;
+
+    protected function setUp(): void
+    {
+        $this->contextBuilder = new CsvEncoderContextBuilder();
+    }
+
+    /**
+     * @dataProvider withersDataProvider
+     *
+     * @param array<string, mixed> $values
+     */
+    public function testWithers(array $values)
+    {
+        $context = $this->contextBuilder
+            ->withDelimiter($values[CsvEncoder::DELIMITER_KEY])
+            ->withEnclosure($values[CsvEncoder::ENCLOSURE_KEY])
+            ->withEscapeChar($values[CsvEncoder::ESCAPE_CHAR_KEY])
+            ->withKeySeparator($values[CsvEncoder::KEY_SEPARATOR_KEY])
+            ->withHeaders($values[CsvEncoder::HEADERS_KEY])
+            ->withEscapedFormulas($values[CsvEncoder::ESCAPE_FORMULAS_KEY])
+            ->withAsCollection($values[CsvEncoder::AS_COLLECTION_KEY])
+            ->withNoHeaders($values[CsvEncoder::NO_HEADERS_KEY])
+            ->withEndOfLine($values[CsvEncoder::END_OF_LINE])
+            ->withOutputUtf8Bom($values[CsvEncoder::OUTPUT_UTF8_BOM_KEY]);
+
+        $this->assertSame($values, $context->toArray());
+    }
+
+    /**
+     * @return iterable<array{0: array<string, mixed>|}>
+     */
+    public function withersDataProvider(): iterable
+    {
+        yield [[
```

Comment:
```php
yield ['With context' => [ CsvEncoder::DELIMITER_KEY => ';',
            CsvEncoder::ENCLOSURE_KEY => '"',
            CsvEncoder::ESCAPE_CHAR_KEY => '\\',
            CsvEncoder::KEY_SEPARATOR_KEY => '_',
            CsvEncoder::HEADERS_KEY => ['h1', 'h2'],
            CsvEncoder::ESCAPE_FORMULAS_KEY => true,
            CsvEncoder::AS_COLLECTION_KEY => true,
            CsvEncoder::NO_HEADERS_KEY => false,
            CsvEncoder::END_OF_LINE => 'EOL',
            CsvEncoder::OUTPUT_UTF8_BOM_KEY => false,
        ]];

Reply:
same for others dataprovider

Reply:
Yes good idea! (it's `yield 'With context' => [[...]]` though)